### PR TITLE
A get at the NamespaceManager does not persist the result anymore

### PR DESCRIPTION
**Pull request by koppor**
We often have changes at `Namespaces.properties` during usage of Winery. This reduces it to no automatic changes.

Feature: More readable prefixes.

Feature: Also handle the NULL Namespace (https://github.com/eclipse/winery/pull/242/commits/c5ff12e25d77d37c682f1ccb09b4fd419f4efb54)

Had also to adapt the test cases to reflect the new prefix generation (3e9f12dda4b89367ab6f67fc071d098436a3dff8)

This PR also fixes modification of the color when just reading the color (https://github.com/eclipse/winery/pull/242/commits/4378175f6286d65a9fab13b063a0e56ef5c2993c).

### CODE REVIEW

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/NamespaceManager.java [hunk 1]
```diff
@@ -13,44 +13,50 @@
...
+
         String ns = namespace.getDecoded();
         return this.getPrefix(ns);
     }
 
     /**
-     * SIDEFFECT: URI is added to list of known namespaces if it did not exist
-     * before
+     * Returns a prefix for the given namespace. With two different namespaces, to different prefixes are returned.
+     * The returned prefixes are not persistest. Thus, two instances of a NamespaceManager might return different
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/NamespaceManager.java]: Why is that? Wouldn't this result in failing tests at some point in time?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java [hunk 2]
```diff
@@ -36,64 +47,149 @@ public ConfigurationBasedNamespaceManager(Configuration configuration) {
...
+            this.namespaceToPrefixMap.remove(namespace);
+        }
+    }
+
+    /**
+     * Generates a string indicating the kind of definitions children maintained by the namespace
+     *
+     * @return empty string if nothing could be matched
+     */
+    private String generateKindString(String namespace) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java]: I would rename it to something like `getDefinitionsTypeAbbreveation()`.
`generateKindString()` is too generic  IMHO.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java [hunk 3]
```diff
@@ -36,64 +47,149 @@ public ConfigurationBasedNamespaceManager(Configuration configuration) {
...
+        String mid;
+        if (namespace.contains("servicetemplates/")) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java]: What if the namespace is `null`?

Comment (koppor) [reply org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java]: Call graph: no null is ever passed. Nevertheless, I added Objects.requireNonNull(namespace)

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/NamespaceManager.java (hunk 1 above)
Comment (koppor) [reply org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/NamespaceManager.java]: f2f discussion: Single instance vs. changes at `Namespaces.properties`. Current decision. Namespace.properties should take the explicitly user-approved namespaces only. Not the implicitly "passt scho" namespaces.

**Pull request merged**